src/main/java/uk/gov/hmcts/ccd/domain/service/caseaccess/CaseAccessOperation.java

```diff
@@ -75,7 +75,7 @@ public void revokeAccess(final String jurisdictionId, final String caseReference
     public List<String> findCasesUserIdHasAccessTo(final String userId) {
         return caseUserRepository.findCasesUserIdHasAccessTo(userId)
             .stream()
-            .map(databaseId -> caseDetailsRepository.findById(databaseId).getReference() + "")
+            .map(databaseId -> String.valueOf(caseDetailsRepository.findCaseDetailsReferenceById(databaseId)))
```

Comment:
Well, this will still fetch a case id for every case the user has access to. So, 40 DB queries for 40 cases.

I was thinking a query with an IN operator would be a more scalable solution.